WcaOnRails/spec/factories/competitions.rb

```diff
@@ -86,6 +86,7 @@
     trait :with_valid_submitted_results do
       announced
       with_rounds { true }
+      results_submitted_at { Time.now }
```

Comment:
Is there a specific reason why you dont't refer to the `results_posted` trait here, similar to the reference to `announced` two lines above?

Reply:
We only need results to be _submitted_, not necessarily _posted_, for the report to be submitted/posted. So just using the trait `:results_posted` in the test itself won't make it pass.

On a side note, it might be worth adding `results_submitted_at { ... }` to the `:results_posted` trait.

Reply:
AH! I completely missed the fact that there's a difference between `results_submitted_at` and `results_posted_at`. Sorry, my bad, your explanation makes sense! But then we have to reconsider the change: Posting a Delegate Report has to work even after the Results were *posted*. Does your code support that case?

Reply:
I changed it to check whether results have been submitted or posted, just in case it's possible for results to be posted without being submitted.

I also realized that helper functions already exist for exactly these purposes, so I'm using those now. Although it seems odd that one is defined via `![...].nil?` and the other via `[...].present?`, given that these aren't the same in edge cases. (See [here](https://github.com/thewca/worldcubeassociation.org/blob/be4031eaf3e28aba027225f3823801441c271d85/WcaOnRails/app/models/competition.rb#L1147), and a few definitions below.)